src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -395,15 +400,20 @@ public ThreadPoolWorkQueue()
             ThreadPoolWorkQueueThreadLocals.threadLocals ??
             (ThreadPoolWorkQueueThreadLocals.threadLocals = new ThreadPoolWorkQueueThreadLocals(this));
 
+        internal bool ThreadRequestNeeded(int count) => (count < ThreadPoolGlobals.processorCount) &&
+            ((count < workItems.Count) || (WorkStealingQueueList.wsqActive > 0));
```

Comment:
It looks like accessing the ConcurrentQueue's Count would freeze the tail if there are at least two segments, and once frozen it looks like it would not be unfrozen and it would have to create a new segment for new items. I'm a bit concerned about accessing Count too often. The largest impact of this change should be to remove the redundant thread requests from Dispatch to avoid waking additional threads when the work item count is low, and I think the first condition (`count < ThreadPoolGlobals.processorcount`) still accomplishes that. My preference is to go with that for now.